src/models/GoalItem.ts

```diff
@@ -4,12 +4,14 @@ export enum Repeat {
 }
 
 export interface GoalItem {
-  id?: number;
+  id?: Number;
   title: string;
   duration: Number;
-  sublist: string[] | null;
+  sublist?: GoalItem[];
```

Comment:
@pulkitvyas08 are you sure it should have GoalItem[] Interface?
@tijlleenders in the sublist is there something other than a string?  Acc to figma board like if I want to go shopping eve1y Wednesday then the sublist will have the items of it right?

Reply:
Yes, refer to the attached snippet. All the sublist items can be deleted, archived, and have their title updated

![image](https://user-images.githubusercontent.com/54947769/175034639-126d6bd0-b4e1-403f-8fe7-a13cc3979c2a.png)

Reply:
cool then. I think we need better example though because scenarios like above i dont see use of other attributes

Reply:
The only one I see not being used is the plus icon. As sublist items cannot have their own sublist. Rest are good

Reply:
The idea is that any list can have a subList, including subLists.
It's a Directed Acyclical Graph hierarchy that can go x levels deep.
That's why I propose separate goals, relationships and properties repositories.